### PR TITLE
Add default values to payloads

### PR DESCRIPTION
**Pull request by ldgeo**
when a key is missing from the payload the sql request should see a NULL value instead of throwing a KeyError

### CODE REVIEW

File: api_li3ds/app.py [hunk 1]
```diff
@@ -13,6 +15,19 @@ class Resource(OrigResource):
...
 
+def get_none():
+    pass
+
+
+def defaultpayload(payload):
+    """Use a default dict to add a None value
+    and avoid a KeyError on sql request interpolation
+    """
+    newpayload = defaultdict(get_none)
```
Comment (elemoine) [inline api_li3ds/app.py]: You could do `defaultdict(lambda: None)` instead.

**Comment by elemoine** [comment]
Let me test it…

**Comment by elemoine** [comment]
> Let me test it…

It kinda works. I removed `validity_start` and `validity_end` from the payload and the insertion of the transfo succeeded. But I have no validity dates for this transfo in the database, the default values that you added to the model did not apply. Is this expected?

**Review by mbredif — COMMENTED** [review]
that looks good. I have not tested it though

**Comment by ldgeo** [comment]
@elemoine if you merge this : https://github.com/LI3DS/pg_li3ds/pull/1 you'll have a default value filled by pg

**Comment by elemoine** [comment]
> @elemoine if you merge this : LI3DS/pg_li3ds#1 you'll have a default value filled by pg

In that case, don't we want to remove the default values set in Python?

File: api_li3ds/app.py (hunk 1 above)
Comment (ldgeo) [reply api_li3ds/app.py]: yes, feel free to change it

**Review by elemoine — APPROVED** [review]
I added a commit that removes the `get_none` function and uses a `lambda` instead.

Feel free tto merge this.

**Pull request merged**